services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationService.kt

```diff
@@ -120,6 +128,15 @@ open class ApplicationService(
             "Invalid geometry received when updating application for user $userId, id=${application.id}, alluid=${application.alluid}, reason = ${validationError.reason}, location = ${validationError.location}"
         }
 
+        val hankeId = application.hanke.id!!
+        application.applicationData.areas?.let { areas ->
```

Comment:
This is checking for existing areas and NOT the new ones. This is an update function and it should verify newApplicationData.